src/test/java/com/github/krzysiekjodlowski/ox/BoardValidatorHorizontalCheckTest.java

```diff
@@ -0,0 +1,219 @@
+
+        //assert
+        assertTrue(isGameOver);
+    }
+
+    @Test(dataProvider = "winningHorizontallyWith4On4x4")
+    public void testHorizontalGameWinWhenConditionIs4On4x4Board(int[] chosenFields) throws NumberLowerThanOneException {
+        //arrange
+        Board board = new Board(4);
+        BoardValidator boardValidator = new BoardValidator(board, 4);
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[0]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[0]), Symbol.CROSS));
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[1]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[1]), Symbol.CROSS));
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[2]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[2]), Symbol.CROSS));
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[3]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[3]), Symbol.CROSS));
+
+
+        //act
+        boolean isGameOver = boardValidator.saysItsOver();
+
+        //assert
+        assertTrue(isGameOver);
+    }
+
+    @Test(dataProvider = "markRowOfThreeFieldsBeginningFromSecond")
+    public void testHorizontalGameWinWhenConditionIs3nTheMiddleOf5x5BoardRow(int[] chosenFields) throws NumberLowerThanOneException {
+        //arrange
+        Board board = new Board(5);
+        BoardValidator boardValidator = new BoardValidator(board, 3);
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[0]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[0]), Symbol.CROSS));
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[1]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[1]), Symbol.CROSS));
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[2]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[2]), Symbol.CROSS));
+
+        //act
+        boolean isGameOver = boardValidator.saysItsOver();
+
+        //assert
+        assertTrue(isGameOver);
+    }
+
+    @Test(dataProvider = "markRowOfFirstMiddleAndLast40FieldsOf40x40Board")
+    public void testHorizontalGameWinWhenConditionIs40On40x40Board(int[] chosenFields) {
+        //arrange
+        Board board = new Board(40);
+        BoardValidator boardValidator = new BoardValidator(board, 40);
+
+        Arrays.stream(chosenFields).forEach(x ->
+                {
+                    try {
+                        board.handle(new Move(FieldNumber.valueOf(x), Symbol.CROSS));
+                        boardValidator.handle(new Move(FieldNumber.valueOf(x), Symbol.CROSS));
+                    } catch (NumberLowerThanOneException e) {
+                        e.printStackTrace();
+                    }
+                }
+        );
```

Comment:
?

Reply:
I had no other way, you can't add throws claue to method signature when using streams:/